clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManager.java

```diff
@@ -21,53 +21,76 @@
 import java.util.Optional;
 
 /**
- * Manages group membership for a single member.
+ * Manages membership of a single member to a consumer group.
+ * <p/>
  * Responsible for:
  * <li>Keeping member state</li>
  * <li>Keeping assignment for the member</li>
  * <li>Computing assignment for the group if the member is required to do so<li/>
  */
 public interface MembershipManager {
 
+    /**
+     * ID of the consumer group the member is part of (or wants to be part of).
+     */
     String groupId();
 
+    /**
+     * Instance ID used by the member when joining the group. If non-empty, it will indicate that
+     * this is a static member.
+     */
     Optional<String> groupInstanceId();
 
+    /**
+     * Member ID assigned by the server to this member when it joins the consumer group.
+     */
     String memberId();
 
+    /**
+     * Current epoch of the member, maintained by the server.
+     */
     int memberEpoch();
 
+    /**
+     * Current state of this member a part of the consumer group, as defined in {@link MemberState}.
```

Comment:
Maybe "the current state of the consumer" because it might not be in a group right?

Reply:
Done. Just for the record, the idea of the membershipManager supporting a NOT_IN_GROUP state does not exist here, but it is introduced in the other [PR](https://github.com/apache/kafka/pull/14390) still under review.